src/codons.py

```diff
@@ -36,8 +36,13 @@ def split_codons(dna: str) -> list[str] | None:
 
     """
     # FIXME: Implement the function
-    return []
-
+    codons = []
+    for s in range(0,len(dna),3):
+        codons.append(dna[s:s+3])
+    for codon in codons:
+        if len(codon)!=3:
+            return None
```

Comment:
You are running through the `codons` list just to check if any codon has a length different from three, but the way you constructed the codons, this can only happen for the very last. It is much faster to just look at the last element than to check _all_ of them.

Of course, `len(dna) % 3 != 0` would also tell you if it is a valid open reading frame, before you run through the string the first time.